### DISCUSSION TITLE
Latest version does not work with new Prism 8.1.97

### DISCUSSION THREAD

**Discussion by vardansargsyan92** [dansiegel/Prism.Plugin.Popups#221] [Q&A]
HI. I am using pop-ups in my application. I have updated the prism packages to 8.1.97 version and now the initial navigation hangs when the application starts.

I have the following classes:

```
    public interface IStaleMonitorNavigationService : INavigationService
    {
        Task ForceGoBackAsync(NavigationParameters parameters = null);
        Task<bool> ShouldDiscard();
    }
```

```
    public class StaleMonitorNavigationServiceDecorator : PopupPageNavigationService, IStaleMonitorNavigationService
    {
        private readonly IPageDialogService _dialogService;

        public StaleMonitorNavigationServiceDecorator(IPopupNavigation popupNavigation,
                                                      IContainerExtension container,
                                                      IApplicationProvider applicationProvider,
                                                      IPageBehaviorFactory pageBehaviorFactor,
                                                      IPageDialogService dialogService) : base(popupNavigation, container, applicationProvider, pageBehaviorFactor)
        {
            _dialogService = dialogService;
        }

        public override async Task<INavigationResult> GoBackAsync(INavigationParameters parameters)
        {
            if(!await ShouldDiscard())
            {
                return new NavigationResult
                       {
                           Success = false
                       };
            }

            INavigationResult result = await base.GoBackAsync(parameters);
            return result;
        }

        public Task ForceGoBackAsync(NavigationParameters parameters = null)
        {
            return base.GoBackAsync(parameters);
        }

        public async Task<bool> ShouldDiscard()
        {
            if(PageExtensions.TopPage(_popupNavigation, _applicationProvider)?.BindingContext is IStaleMonitorViewModel viewModel) return await viewModel.ConfirmDiscard(_dialogService) != false;

            return true;
        }
    }
```
And here is a dependency registration:

```       
 public static IContainerRegistry RegisterFormsDependencies(this IContainerRegistry registry)
        {
		  ...
		  
            // Support pages
            registry.RegisterForNavigation<NavigationPage>();
            registry.RegisterPopupNavigationService<StaleMonitorNavigationServiceDecorator>();
            registry.RegisterSingleton<INavigationService, StaleMonitorNavigationServiceDecorator>();
            registry.RegisterInstance<IStaleMonitorNavigationService>(registry.GetContainer()
			.Resolve<StaleMonitorNavigationServiceDecorator>());

           return registry;
        }
```

When the application hangs in the `StaleMonitorNavigationServiceDecorator` 's constructor and I am not able to do anything. So what is wrong here? Do I need to wait until a new Plugin.Popups release?

**Comment by dansiegel** [comment]
I haven't tested it so I couldn't tell you... that said I only this morning did an update for 8.1.. and there are some things like handling the back button that won't be fully compatible with 8.1 until I get to it...

From looking at your code it seems you are likely duplicating IConfirmNavigation

**Reply by vardansargsyan92** [reply]
hi @dansiegel . Thank you for your response.

It seems to having cyclic dependency injection inside `PopupPageNavigationService` but I am not sure. The thing that is clear for me is when I try to navigate to any page when app starts, the app is going infinity loop inside `PopupPageNavigationService`'s constructor. When I remove custom `PopupPageNavigationService` from container registration then the app is working fine. But in that case I am not able to pass parameter when the pop-up page is dismissed.

**Reply by herrquark** [reply]
Can confirm: `OnNavigatedTo` is not called on parent page after popup was dismissed via `GoBackAsync`

**Reply by vardansargsyan92** [reply]
hi @herrquark . Yes you are right. And it does not matter I am using custom `PopupPageNavigationService ` or no. I am trying to download the source code and understand what is wrong there.

**Reply by vardansargsyan92** [reply]
@dansiegel. Happy to see that you have updated the prism in the Prism.Plugin.Popups

![image](https://user-images.githubusercontent.com/19687305/119820495-94aae680-bf02-11eb-801e-00568b9ff502.png)

I have downloaded the source code and now the `GoBackAsync` works as expected. One question. Now I am not able to do PR-s because the access to master branch is denied. So should I fork the project to make a PR-s or there is other options to do that?

**Reply by dansiegel** [reply]
> @dansiegel. Happy to see that you have updated the prism in the Prism.Plugin.Popups
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/19687305/119820495-94aae680-bf02-11eb-801e-00568b9ff502.png)
> 
> 
> 
> I have downloaded the source code and now the `GoBackAsync` works as expected. One question. Now I am not able to do PR-s because the access to master branch is denied. So should I fork the project to make a PR-s or there is other options to do that?

The "update" as of now is purely a version bump. There are still things that need to be done. 
As for the bug, if you create a fork and push changes to your fork, GitHub will ask if you want to PR the changes back to this repo. I am always happy to accept PRs, too few developers offer.
